### ISSUE TITLE
[Bug] Copy as HTML does not paste as HTML in Texteditor

### ISSUE THREAD

**Issue by BsNoSi** [Zettlr/Zettlr#2635]
### Description

`Edit` → `Copy as HTML` does not copy html into clipboard.

- Pasting clipboart into LibreOffice Writer structure and markup is shown
- pasting clipboard into editor (Notepad++, …) only text is inserted

**This might be an enhancement request.** 

But I'd expect HTML as HTML in clipboard, which means "HTML markup". Current behaviour is pasting the same as standard copy – including Markdown markup (not stripped to be valueble as "text only").

### Reproducing

Mark text with markdown markup → `Edit` → `Copy as HTML`  paste into any textprocessor (✔ markup is shown), paste it into text editor (✗ insert equals markdown of Zettlr)

### Zettlr Version

Stable (most recent version)

### Specify version

2.0.0

### Your Platform

- [X] Windows
- [ ] macOS
- [ ] Linux

### Architecture

- [X] Intel 64bit
- [ ] ARM 64bit

### Operating System Version

Windows 10pro x64

**Comment by nathanlesage** [comment]
Just tested it out and the behavior is as expected: If you paste the text copied via "Copy as HTML" into Word or LibreOffice or any other RichText editor, you see the Markdown converted to actual italics/bold/headings/etc. If you paste it into a text-only editor you get the Markdown source. This is basically the way copying from LibreOffice or other editors works.

**Comment by BsNoSi** [comment]
Ah – understand. 
Then this are two different things.

1. It is not possible, to copy **HTML** as HTML. I can copy **styles** from and to Zettlr.
2. I can copy and **paste styles** from and to word processors that are able to *interpret* HTML held as that in the clipboard.

From this point of view the naming, keyboard shortcuts and connected behaviour look inconsistent to me and should be aligned:

- `Copy` → “text format”, *without* Markdown, *text only* → `Ctrl` `Alt` `C`
- `Copy with style` → “HTML format” for exchange with word processors and “as is” including Markdown markup for text editors  → Default `Ctrl` `C`
- `Paste` → as "clean text format" without style, *text only* → `Ctrl` `Alt` `V`
- `Paste with style` → If available, convert styles into Markdown  → default `Ctrl` `V`

Currently, the functionality is “over cross” and keyboard shortcuts are mixed up with Ctrl [Alt|Shift] Key

My wish, “copy HTML”, is currently not supported, but could be, if this format would be pasted besides “HTML” format as a second format “as text” into the clipboard. But this should be triggered by `Ctrl` `Shift` `C` to align with the logic above.

**Comment by BsNoSi** [comment]
As far as I know other programs, `Ctrl` `C` | `Ctrl` `V` for copying/pasting *formatted* content (if available) has become the quasi-standard, as well as `Ctrl` `Shift` `C` | `Ctrl` `Shift` `V` for “content only”, without formatting. This is the reason, why I stepped into this.

**Comment by mattes3** [comment]
> Just tested it out and the behavior is as expected: If you paste the text copied via "Copy as HTML" into Word or LibreOffice or any other RichText editor, you see the Markdown converted to actual italics/bold/headings/etc. If you paste it into a text-only editor you get the Markdown source. This is basically the way copying from LibreOffice or other editors works.

This was hard to grasp for me. I thought I would actually get my clipboard filled with a stream of characters in hypertext markup language that I can paste into any program I like. I expected it to always show up as hypertext markup language syntax. Turns out it only works if the target program recognizes the HTML content type in the clipboard.

**Comment by nathanlesage** [comment]
Yes, the clipboard concepts are somewhat weird. There are five different clipboard types which need to be accomodated for:

* A `plain text` clipboard: This should always contain some "raw" representation of text without any formatting
* A `HTML` clipboard: This should contain the same text as the plain text clipboard, but with HTML formatting
* A `RTF` clipboard, which does the same thing as HTML, but with RichText formatting
* An `Image` clipboard, which contains image data

This is intended so that if you — for example — copy some text from your Webbrowser, the browser will provide two versions of what you copied: A plain text version without formatting, and an HTML version. If you now paste this content into a plain text editor (such as Notepad or TextEdit) these programs can just read the plain text clipboard, and insert it. If you paste this content into an editor that does support formatting, however, that editor can instead read the HTML clipboard instead of the text clipboard, and possibly transform the HTML structure to — in the case of Zettlr — Markdown.

I named the menu item `Copy as HTML` because I want to stick to the "truth", even if it may be counterfactual. But if you want to see it like that, "Copy as HTML" is semantically exactly equal to "Copy with Formatting" menu items in other software; i.e. it does the exact same thing.
